pkgs/applications/editors/emacs/elisp-packages/idris2-mode/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, trivialBuild
+, fetchFromGitHub
+, emacs
+, emacs28Packages
+}:
+
+trivialBuild rec {
+  pname = "idris2-mode";
+  version = "1.1";
+
+  src = fetchFromGitHub {
+    owner = "idris-community";
+    repo = pname;
+    rev = version;
+    hash = "sha256-rTeVjkAw44Q35vjaERs4uoZRJ6XR3FKplEUCVPHhY7Q=";
+  };
+
+  buildInputs = [
+    emacs
+  ] ++ propagatedUserEnvPkgs;
+
+  propagatedUserEnvPkgs = with emacs.pkgs; [
+    prop-menu
+  ];
```

Comment:
Please use callPackage style dependency injection and not `with emacs.pkgs; ...`.

Reply:
Did you mean like this?